niworkflows/interfaces/mni.py

```diff
@@ -231,18 +354,113 @@ def _validate_results(self):
 
 
 def mask(in_file, mask_file, new_name):
+    """
+    Apply a binary mask to an image.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to a NIfTI file to mask
+    mask_file : str
+        Path to a binary mask
+    new_name : str
+        Path/filename for the masked output image.
+
+    Returns
+    -------
+    str
+        Absolute path of the masked output image.
+
+    Notes
+    -----
+    in_file and mask_file must be in the same
+    image space and have the same dimensions.
+    """
     import nibabel as nb
     import os
-
+    # Load the input image
     in_nii = nb.load(in_file)
+    # Load the mask image
     mask_nii = nb.load(mask_file)
+    # Set all non-mask voxels in the input file to zero.
     data = in_nii.get_data()
     data[mask_nii.get_data() == 0] = 0
+    # Save the new masked image.
     new_nii = nb.Nifti1Image(data, in_nii.affine, in_nii.header)
     new_nii.to_filename(new_name)
     return os.path.abspath(new_name)
 
 
+def create_cfm(in_file, lesion_mask=None, global_mask=True, out_path=None):
+    """
+    Create a mask to constrain registration.
+
+    Parameters
+    ----------
+    in_file : str
+        Path to an existing image (usually a mask).
+        If global_mask = True, this is used as a size/dimension reference.
+    out_path : str
+        Path/filename for the new cost function mask.
+    lesion_mask : str, optional
+        Path to an existing binary lesion mask.
+    global_mask : bool
+        Create a whole-image mask (True) or limit to reference mask (False)
+
+    Returns
+    -------
+    str
+        Absolute path of the new cost function mask.
+
+    Notes
+    -----
+    in_file and lesion_mask must be in the same
+    image space and have the same dimensions
+    """
+    import os
+    import numpy as np
+    import nibabel as nb
+    from nipype.utils.filemanip import fname_presuffix
+
+    if out_path is None:
+        out_path = fname_presuffix(in_file, suffix='_cfm', newpath=os.getcwd())
+    else:
+        out_path = os.path.abspath(out_path)
+
+    if not global_mask and not lesion_mask:
+        NIWORKFLOWS_LOG.warning(
+            'No lesion mask was provided and global_mask not requested, '
+            'therefore the original mask will not be modified.')
+
+    # Load the input image
+    in_img = nb.load(in_file)
+
+    # If we want a global mask, create one based on the input image.
+    if global_mask:
+        # Create a mask of ones with the shape of the input image.
+        data = np.ones(in_img.shape, dtype=np.uint8)
+    else:
+        data = in_img.get_data()
```

Comment:
I think this is assuming that the data in `in_img` is all 0 or 1. Should we make this:

```Python
data = np.asanyarray(in_img.get_data() > 0, dtype=np.uint8)
```

Reply:
Why `asany`?

Reply:
`asanyarray` passes through arrays that don't need to change without copying. `np.array()` of a `np.memmap` will copy, but `np.asanyarray` will not unless it has to (e.g. the dtype doesn't match).

But anyway, we settled with emitting an error rather than silently interpreting a non-mask as a mask.